pageserver/src/http/routes.rs

```diff
@@ -330,7 +330,7 @@ async fn get_lsn_by_timestamp_handler(request: Request<Body>) -> Result<Response
 
     let result = match result {
         LsnForTimestamp::Present(lsn) => format!("{lsn}"),
-        LsnForTimestamp::Future(_lsn) => "future".into(),
+        LsnForTimestamp::Future(lsn) => format!("{lsn}"),
```

Comment:
It would be nice for the caller to still be able to distinguish between the Present and Future cases. I propose exposing the full details:

```
    let result = match result {
        LsnForTimestamp::Present(lsn) => format!("Present({lsn})"),
        LsnForTimestamp::Future(lsn) => format!("Future({lsn})"),
        LsnForTimestamp::Past(lsn) => "Past({lsn})".into(),
        LsnForTimestamp::NoData(lsn) => "NoData({lsn})".into(),
    };
```
(Or some similar format is convenient to parse)

Reply:
This can be done without breaking backwards compatibility by control plane selecting which version of the response they want using `Accept:` header:
- missing or `text/plain`: old format
- `application/json`: new format